staging/src/k8s.io/apimachinery/pkg/util/mergepatch/errors.go

```diff
@@ -21,9 +21,23 @@ import (
 	"fmt"
 )
 
-var ErrBadJSONDoc = errors.New("Invalid JSON document")
-var ErrNoListOfLists = errors.New("Lists of lists are not supported")
-var ErrBadPatchFormatForPrimitiveList = errors.New("Invalid patch format of primitive list")
+var (
+	ErrBadJSONDoc                     = errors.New("Invalid JSON document")
+	ErrNoListOfLists                  = errors.New("Lists of lists are not supported")
+	ErrBadPatchFormatForPrimitiveList = errors.New("Invalid patch format of primitive list")
+)
+
+func ErrNoMergeKey(m map[string]interface{}, k string) error {
+	return fmt.Errorf("map: %v does not contain declared merge key: %s", m, k)
+}
+
+func ErrBadArgType(expected, actual string) error {
+	return fmt.Errorf("expected a %s, but received a %s", expected, actual)
```

Comment:
I wonder if it would simplify the code to take `expected, actual interface{}` and the print the type with `%T` (instead of using `reflect.TypeOf(obj).Kind().String()`).